tests/basic/pure_test.go

```diff
@@ -335,3 +337,108 @@ var _ = Describe("{BringUpLargePodsVerifyNoPanic}", func() {
 		AfterEachTest(contexts, testrailID, runID)
 	})
 })
+
+// This test Creates multiple FADA volume/app (nginx) - Restart PX on a Node while volume creation is in progress
+/*
+https://portworx.testrail.net/index.php?/tests/view/72615026
+
+*/
+var _ = Describe("{RestartPXWhileVolCreate}", func() {
+	var contexts []*scheduler.Context
+	var wg sync.WaitGroup
+	JustBeforeEach(func() {
+		StartTorpedoTest("RestartPXWhileVolCreate", "Test creates multiple FADA volume and restarts px on a node while volume creation is in progress", nil, 72615026)
+	})
+	It("schedules nginx fada volumes on (n) * (NumberOfDeploymentsPerRestart) different namespaces and restarts portworx on a different node after every NumberOfDeploymentsPerRestart have been queued to schedule", func() {
+		//Provisioner for pure apps
+		Provisioner := fmt.Sprintf("%v", portworx.PortworxCsi)
+		//Number of times portworx has to be restarted
+		n := 3
+		//Number of apps to be deployed after which a restart can be triggered
+		NumberOfDeploymentsPerRestart := 15
+		//Restart portworx n number of times
+		stepLog = "start provisioning nginx apps in the created namespaces and for every NumberOfDeploymentsPerRestart restart portworx on one of the nodes"
+		Step(stepLog, func() {
+			for i := 0; i < n; i++ {
+				// Step 1: Schedule applications
+				wg.Add(1)
+				go func() {
+					defer wg.Done()
+					Step("Schedule applications", func() {
+						log.InfoD("Scheduling applications")
+						for j := 0; j < NumberOfDeploymentsPerRestart; j++ {
+							taskName := fmt.Sprintf("test%v", (j)+NumberOfDeploymentsPerRestart*i)
+							context, err := Inst().S.Schedule(taskName, scheduler.ScheduleOptions{
+								AppKeys:            Inst().AppList,
+								StorageProvisioner: Provisioner,
+								PvcSize:            6 * units.GiB,
+							})
+							log.FailOnError(err, "Failed to schedule application of %v namespace", taskName)
+							contexts = append(contexts, context...)
+						}
+					})
+				}()
+
+				// Step 2: Restart Portworx
+				wg.Add(1)
+				go func() {
+					defer wg.Done()
+					stepLog := "Restart Portworx"
+					Step(stepLog, func() {
+						nodes := node.GetStorageDriverNodes()
+						randomNode := rand.Intn(len(nodes))
```

Comment:
how are you making sure, node where volume is getting created in restarted.
you might need to use volume placement strategies for this

Reply:
Can we inspect volume and do a restart based on response?

Reply:
Picking of random nodes was suggested by hari. So I went ahead with it. Maybe we can discuss this further.

Reply:
Done with placing volumes in the selected node and restarting px in that particular node.